src/jit/lowerxarch.cpp

```diff
@@ -2104,20 +2105,48 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* putArgStk)
             const unsigned  fieldOffset = current->gtFieldOffset;
             assert(fieldType != TYP_LONG);
 
-            // TODO-X86-CQ: we could probably improve codegen here by marking all of the operands to field nodes that
-            // we are going to `push` on the stack as reg-optional.
+            // For x86 we must mark all integral fields as contained or reg-optional, and handle them
+            // accordingly in code generation, since we may have up to 8 fields, which cannot all be in
```

Comment:
>// accordingly in code generation, since we may have up to 8 fields, which cannot all be in [](start = 12, length = 91)

Why "up to 8 fields"? Is that obvious for some reason?